linkml/generators/pydanticgen/array.py

```diff
@@ -316,28 +447,19 @@ def complex_dimensions(self, array: ArrayExpression) -> RangeResult:
                 # equivalent to labeled shape
                 return self.parameterized_dimensions(array)
             else:
-                raise ValueError(
-                    "if exact_number_dimensions is provided, it must be greater than the parameterized dimensions"
-                )
+                # invalid - raise exception
+                ArrayValidator.array_consistent_n_dimensions(array)
 
         elif array.maximum_number_dimensions is not None and not array.maximum_number_dimensions:
             # unlimited n dimensions, so innermost is AnyShape with dtype
             res = self.any_shape(with_inner_union=True)
 
-            if array.minimum_number_dimensions and array.minimum_number_dimensions > len(array.dimensions):
+            if array.minimum_number_dimensions:
                 # some minimum anonymous dimensions but unlimited max dimensions
                 # e.g., if min = 3, len(dim) = 2, then res.range = List[Union[AnyShapeArray[dtype], dtype]]
                 # res.range will be wrapped with the 2 labeled dimensions later
                 res.range = self._list_of_lists(array.minimum_number_dimensions - len(array.dimensions), res.range)
 
-        elif array.minimum_number_dimensions and array.maximum_number_dimensions is None:
-            raise ValueError(
-                (
-                    "Cannot specify a minimum_number_dimensions while maximum is None while using labeled dimensions - "
-                    "either use exact_number_dimensions > len(dimensions) for extra parameterized dimensions or set "
-                    "maximum_number_dimensions explicitly to False for unbounded dimensions"
-                )
-            )
```

Comment:
in line with above, would want to be consistent, so if we leave errors in then we might as well bring these back and use the validator to raise errors, or else remove them all and leave some holes in the switch statements